tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnectionTests.cs

```diff
@@ -53,18 +53,18 @@ public BinaryConnectionTests()
 
             _endPoint = new DnsEndPoint("localhost", 27017);
             var serverId = new ServerId(new ClusterId(), _endPoint);
+            var connectionId = new ConnectionId(serverId);
+            var isMasterResult = new IsMasterResult(new BsonDocument { { "ok", 1 }, { "maxMessageSizeBytes", 48000000 } });
+            var buildInfoResult = new BuildInfoResult(new BsonDocument { { "ok", 1 }, { "version", "2.6.3" } });
+            var connectionDescription = new ConnectionDescription(connectionId, isMasterResult, buildInfoResult);
 
             _mockConnectionInitializer = new Mock<IConnectionInitializer>();
-            _mockConnectionInitializer.Setup(i => i.InitializeConnection(It.IsAny<IConnection>(), CancellationToken.None))
-                .Returns(() => new ConnectionDescription(
-                    new ConnectionId(serverId),
-                    new IsMasterResult(new BsonDocument()),
-                    new BuildInfoResult(new BsonDocument("version", "2.6.3"))));
-            _mockConnectionInitializer.Setup(i => i.InitializeConnectionAsync(It.IsAny<IConnection>(), CancellationToken.None))
-                .Returns(() => Task.FromResult(new ConnectionDescription(
-                    new ConnectionId(serverId),
-                    new IsMasterResult(new BsonDocument()),
-                    new BuildInfoResult(new BsonDocument("version", "2.6.3")))));
+            _mockConnectionInitializer
+                .Setup(i => i.InitializeConnection(It.IsAny<IConnection>(), CancellationToken.None))
+                .Returns(connectionDescription);
+            _mockConnectionInitializer
+                .Setup(i => i.InitializeConnectionAsync(It.IsAny<IConnection>(), CancellationToken.None))
+                .Returns(Task.FromResult(connectionDescription));
```

Comment:
`ReturnsAsync`?

Reply:
This was a change that @rstam suggested, but it does look like it should be async here.

Reply:
Not sure when Moq added support for `ReturnAsync` but looks to me like the following two statements are equivalent:

    Returns(Task.FromResult(connectionDescription))

    ReturnsAsync(connectionDescription)

I'm totally fine with changing to `ReturnAsync`. Thanks for finding this method I did not know about!